src/wallet/rpcwallet.cpp

```diff
@@ -5520,17 +5520,15 @@ UniValue z_mergetoaddress(const UniValue& params, bool fHelp)
                 isToTaddr = true;
             },
             [&](libzcash::SaplingPaymentAddress addr) {
-                isToSaplingZaddr = true;
                 // If Sapling is not active, do not allow sending to a sapling addresses.
                 if (!saplingActive) {
                     throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid parameter, Sapling has not activated");
                 }
+                estimatedTxSize += OUTPUTDESCRIPTION_SIZE;
```

Comment:
```suggestion
                estimatedTxSize += OUTPUTDESCRIPTION_SIZE * 2;
```
Personally I would just set `size_t estimatedTxSize = MIN_TX_COST` as I suggested [here](https://github.com/zcash/zcash/pull/6569#discussion_r1173111205), and delete this line and line 5531. That is simpler and a perfectly sufficient estimate for how `estimateTxSize` is used.